conference.js

```diff
@@ -600,76 +600,67 @@ export default {
 
                     return [];
                 });
-        } else if (!requestedAudio && !requestedVideo) {
-            // Resolve with no tracks
-            tryCreateLocalTracks = Promise.resolve([]);
-        } else {
+        } else if (requestedAudio || requestedVideo) {
             tryCreateLocalTracks = createLocalTracksF({
                 devices: initialDevices,
                 timeout,
                 firePermissionPromptIsShownEvent: true
             })
-                .catch(err => {
-                    if (requestedAudio && requestedVideo) {
-
-                        // Try audio only...
-                        errors.audioAndVideoError = err;
+            .catch(async error => {
+                if (error.name === JitsiTrackErrors.TIMEOUT) {
+                    errors.audioAndVideoError = error;
 
-                        if (err.name === JitsiTrackErrors.TIMEOUT && !browser.isElectron()) {
-                            // In this case we expect that the permission prompt is still visible. There is no point of
-                            // executing GUM with different source. Also at the time of writing the following
-                            // inconsistency have been noticed in some browsers - if the permissions prompt is visible
-                            // and another GUM is executed the prompt does not change its content but if the user
-                            // clicks allow the user action isassociated with the latest GUM call.
-                            errors.audioOnlyError = err;
-                            errors.videoOnlyError = err;
+                    return [];
+                }
 
-                            return [];
-                        }
+                // Retry with separate gUM calls.
+                const gUMPromises = [];
+                const tracks = [];
 
-                        return createLocalTracksF(audioOptions);
-                    } else if (requestedAudio && !requestedVideo) {
-                        errors.audioOnlyError = err;
+                if (requestedAudio) {
+                    gUMPromises.push(createLocalTracksF(audioOptions));
+                }
 
-                        return [];
-                    } else if (requestedVideo && !requestedAudio) {
-                        errors.videoOnlyError = err;
+                if (requestedVideo) {
+                    gUMPromises.push(createLocalTracksF({
```

Comment:
I'm wondering shouldn't we wait for the audio GUM to finish before calling the video GUM. If you haven't given permissions and the browser needs to display 2 different popups for permission what happens? Is this a problem?

Reply:
We will not be hitting this condition since the browser displays the permission prompt only once for the first gUM call. If both the devices are not allowed the first time, the subsequent gUM calls will fail immediately (without showing a permission prompt)

Reply:
what happens if I ignore the prompt?

Reply:
Even if you ignore the first prompt (which is not the normal behavior), the audio popup will appear first followed by the video one and I don't see any issues with that.

Reply:
ok. I just wanted to verify you've tested it!